pallets/creditcoin/src/migrations/v6.rs

```diff
@@ -14,14 +19,53 @@ pub use v5::OrderId as OldOrderId;
 pub use v5::Transfer as OldTransfer;
 pub use v5::TransferKind as OldTransferKind;
 
+use crate::Address;
+use crate::AddressId;
+use crate::Blockchain;
 use crate::Transfer;
 use crate::TransferId;
 
+impl TryFrom<OldBlockchain> for Blockchain {
+	type Error = OldBlockchain;
+	fn try_from(old: OldBlockchain) -> Result<Blockchain, OldBlockchain> {
+		match old {
+			OldBlockchain::Ethereum => Ok(Blockchain::ETHEREUM),
+			OldBlockchain::Rinkeby => Ok(Blockchain::RINKEBY),
+			// this assumes that Luniverse == mainnet luniverse, we may want to make the chain ID of the
+			// old "Luniverse" variant on-chain-storage to make testnet work
+			OldBlockchain::Luniverse => Ok(Blockchain::LUNIVERSE),
+			other => Err(other),
+		}
+	}
+}
+
+fn translate_blockchain(old: OldBlockchain) -> Option<Blockchain> {
+	match old {
+		OldBlockchain::Ethereum => Some(Blockchain::ETHEREUM),
+		OldBlockchain::Rinkeby => Some(Blockchain::RINKEBY),
+		// this assumes that Luniverse == mainnet luniverse, we may want to make the chain ID of the
+		// old "Luniverse" variant on-chain-storage to make testnet work
+		OldBlockchain::Luniverse => Some(Blockchain::LUNIVERSE),
+		other => {
+			log::warn!(
+				"unexpected blockchain found on storage item: {:?}",
+				core::str::from_utf8(other.as_bytes()).ok()
+			);
+			None
+		},
+	}
+}
```

Comment:
It looks like both fn's are almost the same. Consider keeping the former, move the log into it, and call Result::ok() on its output whenever you need an option.